src/main/scala/se/lth/immun/TargetDeisotoper.scala

```diff
@@ -1,57 +1,162 @@
 package se.lth.immun
 
 import scala.collection.mutable.ArrayBuffer
+import scala.collection.mutable.Queue
+import scala.collection.mutable.HashSet
+
+import akka.actor._
 
 import se.lth.immun.chem.Peptide
 import se.lth.immun.chem.Constants
 import se.lth.immun.chem.IsotopeDistribution
 
-class TargetDeisotoper(val params:DinosaurParams) extends Timeable {
+class TargetDeisotoper(val params:DinosaurParams) extends Actor with Timeable {
```

Comment:
I don't have the time and energy to verify all the new code here, I'll trust that you've put it to use successfully. I think the targeted mode was very experimental before anyway, so guess this _is_ way better.